src/test/functional/steps/opening-hours.ts

```diff
@@ -0,0 +1,85 @@
+import { Then, When } from 'cucumber';
+import { expect } from 'chai';
+
+import * as I from '../utlis/puppeteer.util';
+
+When('I click the opening hours tab', async () => {
+  const selector = '#tab_opening-hours';
+  const elementExist = await I.checkElement(selector);
+  expect(elementExist).equal(true);
+  await I.click(selector);
+});
+
+Then('I can view the existing opening hours', async () => {
+  const elementExist = await I.checkElement('#openingTimesForm');
+  expect(elementExist).equal(true);
+
+  const tabClosed = await I.checkElement('#opening-hours.govuk-tabs__panel--hidden');
+  expect(tabClosed).equal(false);
+});
+
+When('I enter new opening hours entry', async () => {
+  const inputSelector = '#newOpeningTimeHours';
+  const selectSelector = 'select[name="newOpeningTimeDescription"]';
+
+  const selectElementExists = await I.checkElement(selectSelector);
+  expect(selectElementExists).equal(true);
+  const inputElementExists = await I.checkElement(inputSelector);
+  expect(inputElementExists).equal(true);
+
+  await I.selectItem(selectSelector, '44');
+  await I.fillField(inputSelector, '9am to 5pm');
+});
+
+When('I click the Add button', async () => {
```

Comment:
Can this be reused for the other tabs? I.e set up with an element and a count parameter? Dw if its a pain

Reply:
I've refactored this into a re-usable method in the opening-hours.ts file. If there is a need for re-use, it can now be moved to a separate, common class easily.